lib/verifier/post_zephir_verifier.rb

```diff
@@ -0,0 +1,195 @@
+            "output line count (%s = %s) != input line count (%s = %s)",
+            output_path,
+            output_linecount,
+            input_path,
+            input_linecount
+          )
+        )
+      end
+    end
+
+    # Frequency: ALL
+    # Files: CATALOG_PREP/zephir_upd_YYYYMMDD.json.gz and CATALOG_PREP/zephir_upd_YYYYMMDD_delete.txt.gz
+    #   and potentially CATALOG_PREP/zephir_full_YYYYMMDD_vufind.json.gz
+    # Contents:
+    #   json.gz files: ndj with one catalog record per line
+    #   delete.txt.gz: see `#verify_deletes_contents`
+    # Verify:
+    #   readable
+    #   TODO: deletes file is combination of two component files in TMPDIR?
+    def verify_catalog_prep(date: current_date)
+      delete_file = Derivative::Delete.new(date: date)
+
+      if verify_file(path: delete_file.path)
+        verify_deletes_contents(path: delete_file.path)
+      end
+
+      Derivative::CatalogPrep.derivatives_for_date(date: date).each do |derivative|
+        verify_file(path: derivative.path)
+      end
+    end
+
+    # Verify contents of the given file consists of catalog record IDs (9 digits)
+    # or blank lines
+    def verify_deletes_contents(path:)
+      Zlib::GzipReader.open(path).each_line do |line|
+        if line != "\n" && !line.match?(/^\d{9}$/)
+          error message: "Unexpected line in #{path} (was '#{line.strip}'); expecting catalog record ID (9 digits)"
+        end
+      end
+    end
+
+    # Frequency: DAILY
+    # Files: TMPDIR/vufind_incremental_YYYY-MM-DD_dollar_dup.txt.gz
+    #
+    # Contents:
+    #
+    # Historically, un-dollarized uc1 HTIDs (e.g., uc1.b312920) one per line.
+    # These files originally served as a way to report back to Zephir on items to
+    # "uningest" related to a change that University of California made
+    # regarding certain barcode ranges -- things like uc1.b312920 moved to
+    # uc1.$b312920, and we needed to 'uningest' uc1.b312920.
+    #
+    # Later, it served as a more general way to cause Zephir to mark items as not
+    # ingested and thereby no longer export them in full files. This
+    # functionality (as of 2024) has not been used in many years. If we at some
+    # point begin fully deleting material from the repository, this
+    # functionality could again be used.
+    #
+    # As of December 2024, these files are generated each day, but are expected
+    # to be empty.
+    #
+    # Verify:
+    #  readable
+    #  empty
+    def verify_dollar_dup(date: current_date)
+      dollar_dup = Derivative::DollarDup.new(date: date).path
+      return unless verify_file(path: dollar_dup)
+      gz_count = gzip_linecount(path: dollar_dup)
+      if gz_count.positive?
+        error message: "spurious dollar_dup lines: #{dollar_dup} should be empty (found #{gz_count} lines)"
+      end
+    end
+
+    # Frequency: MONTHLY
+    # Files:
+    #   INGEST_BIBRECORDS/groove_full.tsv.gz
+    #   INGEST_BIBRECORDS/zephir_ingested_items.txt.gz
+    # Contents: TODO
```

Comment:
Are we expecting to do this in the short term? I'm OK not doing that, since I think we're just passing these through from Zephir. OTOH, ingest probably won't give any useful errors if the format is incorrect. See #53

Reply:
`postZephir.pm` is creating `zephir_ingested_items.txt.gz` data and the monthly shell script is compressing/moving to `INGEST_BIBRECORDS` so yeah I think there's a slight opportunity for the format to somehow get borked up. I wouldn't insist on it for the short term, however.

Reply:
Right, so we probably should actually verify the format at least of `zephir_ingested_items.txt.gz`.